external/fv3fit/fv3fit/reservoir/utils.py

```diff
@@ -121,6 +126,30 @@ def process_batch_data(
         return rank_divider.get_all_subdomains_with_flat_feature(data_trimmed)
 
 
+def process_validation_batch_data_to_dataset(
```

Comment:
Reason for a new tfdataset batch processing function: validation using an adapter model needs a xarray dataset as input and there is no need to convert to latent space.